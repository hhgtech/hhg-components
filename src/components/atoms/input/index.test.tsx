import React from 'react'

import { render, cleanup } from '@testing-library/react'
import 'jest-styled-components'

import { Input } from '.'
import MockTheme from './../../../mockTheme'

afterEach(cleanup)

test('it renders', () => {
  const tag = render(
    <MockTheme>
      <Input value="test" name="test-input" size="md" label="test input" />
    </MockTheme>,
  )

  const self = tag.container.firstChild

  expect(self).toMatchSnapshot()
  expect(self).toContainHTML('test input')
  expect(self).toHaveAttribute('data-size', 'md')
})
