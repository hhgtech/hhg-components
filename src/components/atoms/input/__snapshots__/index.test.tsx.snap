// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders 1`] = `
.c0 {
  position: relative;
  min-width: 343px;
  min-height: 114px;
}

.c0[data-has-error] input {
  border-color: #e64728;
}

.c0 input {
  width: 100%;
  padding: 12px 16px;
  border: solid 1px #d9d9d9;
  border-radius: 6px;
}

.c0 input:active,
.c0 input:focus {
  border-color: #2d87f3;
  box-shadow: 0px 0px 2px 2px #91caff;
}

.c0 input:disabled {
  background-color: #f2f2f2;
  cursor: not-allowed;
}

.c0[data-has-action-icon] input {
  padding: 12px 48px 12px 16px;
}

.c0[data-has-display-icon] input {
  padding: 12px 16px 12px 48px;
}

.c0[data-has-action-icon][data-has-display-icon] input {
  padding: 12px 48px;
}

.c0[data-size='lg'] input {
  font-size: 16px;
  font-weight: 400;
}

.c0[data-size='md'] input {
  font-weight: 400;
  line-height: 22px;
}

.c0[data-size='sm'] input {
  font-weight: 400;
  line-height: 20px;
}

.c1 {
  display: inline-block;
  margin-bottom: 4px;
  color: #595959;
  font-weight: 700;
}

@media (max-width:767px) {
  .c0[data-size='lg'] input {
    font-size: 22px;
    line-height: 1.2;
  }
}

@media (min-width:768px) {
  .c0[data-size='lg'] input {
    font-size: 26px;
    line-height: 1.2;
  }
}

@media (max-width:767px) {
  .c0[data-size='md'] input {
    font-size: 14px;
    line-height: 1.2;
  }
}

@media (min-width:768px) {
  .c0[data-size='md'] input {
    font-size: 16px;
    line-height: 1.2;
  }
}

@media (max-width:767px) {
  .c0[data-size='sm'] input {
    font-size: 13px;
    line-height: 1.5;
  }
}

@media (min-width:768px) {
  .c0[data-size='sm'] input {
    font-size: 13px;
    line-height: 1.5;
  }
}

<div
  class="c0"
  data-size="md"
>
  <label
    class="c1 inputLabel"
    for="test-input"
  >
    test input
  </label>
  <div
    style="position: relative;"
  >
    <input
      autocomplete="off"
      id="test-input"
      type="text"
      value="test"
    />
  </div>
</div>
`;
